app/pages/lab/export-workflows-dialog.jsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import apiClient from 'panoptes-client/lib/api-client';
+
+class ExportWorkflowsDialog extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      project_id: -1,
+      workflows: [],
+      workflowSelected: false,
+    };
+
+    this.toggleExport = this.toggleExport.bind(this);
+    this.requestDataExport = this.requestDataExport.bind(this);
+  }
+
+  componentDidMount() {
+    this.updateWorkflowsFromProject(this.props.project);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.updateWorkflowsFromProject(nextProps.project);
+  }
+
+  updateWorkflowsFromProject(project) {
+    if (!project) {
+      return;
+    }
+
+    const order = (project.configuration && project.configuration.workflow_order) || [];
+
+    // TODO: this API call duplicates information fetched to draw the lab sidebar.
+    // when we do the lab refactor, this should be cached somewhere
+    project.get('workflows', { fields: 'display_name' })
+      .then((workflows) => {
+        let result = workflows;
+
+        if (order && order.length > 0) {
+          const dict = {};
+          workflows.forEach((workflow) => { dict[workflow.id] = workflow; });
+          result = order.map((id) => { return dict[id]; });
+        }
+
+        this.setState({ workflows: result });
+      });
+  }
+
+  toggleExport() {
+    if (this.workflowList.selectedIndex >= 0) {
+      this.setState({ workflowSelected: true });
+    }
+  }
+
+  requestDataExport() {
+    try {
```

Comment:
It doesn't make sense to me to use a try/catch block instead of calling a `then()` and `catch()` on the api client promise. Is there a reason to use try/catch here. I think we would want to wait for the `POST` to complete and then catch the error or then perform a function.

This is what I'm describing to do instead: https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/partials/data-export-button.cjsx#L23-L28